ruby/lib/rspec/queue.rb

```diff
@@ -0,0 +1,204 @@
+          queue_config.build_id = build_id
+        end
+
+        help = split_heredoc(<<-EOS)
+          Optional. Sets a prefix for the build id in case a single CI build runs multiple independent test suites.
+            Example: --namespace integration
+        EOS
+        parser.separator ""
+        parser.on('--namespace NAMESPACE', *help) do |namespace|
+          queue_config.namespace = namespace
+        end
+
+        help = split_heredoc(<<-EOS)
+          Specify a timeout after which if a test haven't completed, it will be picked up by another worker.
+          It is very important to set this vlaue higher than the slowest test in the suite, otherwise performance will be impacted.
+          Defaults to 30 seconds.
+        EOS
+        parser.separator ""
+        parser.on('--timeout TIMEOUT', *help) do |timeout|
+          queue_config.timeout = Float(timeout)
+        end
+
+        help = split_heredoc(<<-EOS)
+          A unique identifier for this worker, It must be consistent to allow retries.
+          If not specified, retries won't be available.
+          It's automatically inferred on Buildkite and CircleCI.
+        EOS
+        parser.separator ""
+        parser.on('--worker WORKER_ID', *help) do |worker_id|
+          queue_config.worker_id = worker_id
+        end
+
+        help = split_heredoc(<<-EOS)
+          Defines how many time a single test can be requeued.
+          Defaults to 0.
+        EOS
+        parser.separator ""
+        parser.on('--max-requeues MAX', *help) do |max|
+          queue_config.max_requeues = Integer(max)
+        end
+
+        help = split_heredoc(<<-EOS)
+          Defines how many requeues can happen overall, based on the test suite size. e.g 0.05 for 5%.
+          Defaults to 0.
+        EOS
+        parser.separator ""
+        parser.on('--requeue-tolerance RATIO', *help) do |ratio|
+          queue_config.requeue_tolerance = Float(ratio)
+        end
+
+        parser
+      end
+
+      def split_heredoc(string)
+        string.lines.map(&:strip)
+      end
+
+      def queue_config
+        ::RSpec::Queue.config
+      end
+    end
+
+    RSpec::Core::Parser.prepend(ParserExtension)
+
+    class SingleExample
+      attr_reader :example_group, :example
+
+      def initialize(example_group, example)
+        @example_group = example_group
+        @example = example
+      end
+
+      def id
+        example.id
+      end
+
+      def run(reporter)
+        return if RSpec.world.wants_to_quit
+        instance = example_group.new(example.inspect_output)
+        example_group.set_ivars(instance, example_group.before_context_ivars)
+        succeeded = example.run(instance, reporter)
+        if !succeeded && reporter.fail_fast_limit_met?
+          RSpec.world.wants_to_quit = true
+        end
+        succeeded
+      end
+    end
+
+    URL = 'list:.%2Ftest%2Ffixtures%2Fdummy_spec.rb%5B1%3A1%5D:.%2Ftest%2Ffixtures%2Fdummy_spec.rb%5B1%3A2%5D'
```

Comment:
What is this used for?

Reply:
That's some early test code left over -_-.